chord_metadata_service/discovery/api_views.py

```diff
@@ -139,11 +181,11 @@ async def public_overview(_request: DrfRequest):
     async def _get_field_response(field_id: str, field_props: dict) -> dict:
         stats: list[BinWithValue] | None
         if field_props["datatype"] == "string":
-            stats = await get_categorical_stats(field_props, low_counts_censored=True)
+            stats = await get_categorical_stats(field_props, project_id, dataset_id, low_counts_censored=True)
```

Comment:
this function is typed as `project_id: str` but here `project_id: str | None`.
maybe make the categorical stats function take `str | None` for these and default them to `None` instead, and normalize project/dataset ID getters above with something like `dataset_id = request.query_params.get("dataset") or None`